src/Raven.Server/Monitoring/Snmp/Objects/Database/5.1/DatabaseTotalCount.cs

```diff
@@ -0,0 +1,34 @@
+using System.Linq;
+using Lextm.SharpSnmpLib;
+using Raven.Client;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Monitoring.Snmp.Objects.Documents
+{
+    public class DatabaseTotalCount : ScalarObjectBase<Integer32>
+    {
+        private readonly ServerStore _serverStore;
+
+        public DatabaseTotalCount(ServerStore serverStore)
+            : base("5.1.1")
+        {
+            _serverStore = serverStore;
+        }
+
+        protected override Integer32 GetData()
+        {
+            return new Integer32(GetCount(_serverStore));
+        }
+
+        private static int GetCount(ServerStore serverStore)
+        {
+            using (serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
```

Comment:
This does `O(n)` operation, how often do we expect this to be called, if this is a value being monitored?

Reply:
Most often values are in minutes. Critical endpoints like server up-time are set to low ones like once every minute. But others are not processed that often. Of course I cannot forbid someone from setting the value to 1 second, but I'm also not sure if this is something that we should handle.

Reply:
Okay, if that is the case, we shouldn't worry too much